database/factories/ModelFactory.php

```diff
@@ -22,3 +22,99 @@
         'remember_token' => str_random(10),
     ];
 });
+
+// User with Centre
+$factory->state(App\User::class, 'withRandomCentre', function () {
+
+    $centres  = App\Centre::get();
+
+    if ($centres->count() > 0) {
+        // Pick a random Centre
+        $centre = $centres[random_int(0, $centres->count()-1)];
+    } else {
+        // There should be at least one Centre
+        $centre = factory(App\Centre::class)->create();
+    }
+
+    return [
+        'centre_id' => $centre->id,
+    ];
+});
+
+// Centre
+/** @var \Illuminate\Database\Eloquent\Factory $factory */
+$factory->define(App\Centre::class, function (Faker\Generator $faker) {
+
+    $sponsors = App\Sponsor::get();
+
+    if ($sponsors->count() > 0) {
+        // Pick a random Sponsor
+        $sponsor = $sponsors[random_int(0, $sponsors->count()-1)];
+    } else {
+        // There must be at least one Sponsor
+        $sponsor = factory(App\Sponsor::class)->create();
+    }
+
+    return [
+        'name' => $faker->streetName,
+        'sponsor_id' => $sponsor->id,
+    ];
+});
+
+// Sponsor, or Local Authority.
+$factory->define(App\Sponsor::class, function (Faker\Generator $faker) {
+
+    $counties = [
+        "Barnfordshire",
+        "Barsetshire",
+        "Borsetshire",
+        "Burtondon",
+        "Diddlesex",
+        "Downshire",
+        "Ffhagdiwedd",
+        "Gaultshire",
+        "Glebeshire",
+        "Glenshire",
+        "West PassingBury",
+        "Loamshire",
+        "Mangelwurzelshire",
+        "Markshire",
+        "Mallardshire",
+        "Melfordshire",
+        "Mertonshire",
+        "Mortshire",
+        "Midsomer",
+        "Mummerset",
+        "Naptonshire",
+        "Oatshire",
+        "Placefordshire",
+        "Quantumshire",
+        "Radfordshire",
+        "Redshire",
+        "Russetshire",
+        "Rutshire",
+        "Shiring",
+        "Shroudshire",
+        "Slopshire",
+        "Southmoltonshire",
+        "South Riding",
+        "Stonyshire",
+        "Trumptonshire",
+        "Wessex",
+        "Westershire",
+        "Waringham",
+        "Westshire",
+        "Winshire",
+        "Wordenshire",
+        "Worfordshire",
+        "South Worfordshire",
+        "Wyverndon",
+    ];
+
+    $index = $faker->unique()->numberBetween(0, 43);
```

Comment:
Cant you use `$faker->county` instead of making your own list?

Reply:
I couldn't find county() it in the list faker provides. So I googled some imaginary ones :smile:

Reply:
There is a county. I am sure we use it for LAs in MOMO.

Reply:
You are correct, found it! Stubbonly fond of my version for now though ;-)